plugins/speculation-rules/hooks.php

```diff
@@ -19,6 +19,34 @@
  * @since 1.0.0
  */
 function plsr_print_speculation_rules(): void {
+	// Skip speculative loading for logged-in users.
+	if ( is_user_logged_in() ) {
+		return;
+	}
```

Comment:
Should there also be a filter for this? Maybe I have a membership site and I want to opt-in Speculative Loading.

Alternatively, should this be exposed as an option on the settings screen?

Reply:
For the Core implementation, this will be handled as part of the overall configuration filter, so there it'll be filterable. But here, we would need another extra filter just for this - since we're hopefully close to a core merge, I would prefer not to introduce one just for the plugin.